tests/functional/utils/test_abi.py

```diff
@@ -3,18 +3,46 @@
 
 from ape.utils import parse_type
 
+# An alphanumeric type identifier or array declaration.
+IDENT = r"(\w+(\[\])?)"
+
+# A tuple declaration with one or more items.
+# e.g. (int,) or (int, int) or (int, int,) etc.
+TUPLE = rf"(\({IDENT},(\s*{IDENT},?)*\))"
+
+# An optionally nested tuple.
+# e.g. (int,) or (int, int) or ((int,), int) etc.
+NESTED_TUPLE = rf"(\(({IDENT}|{TUPLE}),(\s*{IDENT}|{TUPLE},?)*\))"
+
+# The full type declaration pattern which matches:
+# - type identifier
+# - array declaration
+# - tuples of type/array
+# - tuples of type/array or tuples
+TYPE_DECL = rf"^({IDENT}|{TUPLE}|{NESTED_TUPLE})$"
+
 
 @pytest.mark.fuzzing
-@given(strategies.from_regex(r"\(*[\w|, []]*\)*"))
+@given(strategies.from_regex(TYPE_DECL))
 def test_parse_type(s):
-    # Example matching strings from above regex:
-    #   * (int, int)
-    #   * ((int, int), int)
-    #   * int
-    #   * uint256
-    #   * int[]
-    #   * (asd ) [] asdf ff 33 asdf
+    # Examples matching strings from above regex:
+    #   - int
+    #   - int[]
+    #   - uint256
+    #   - (int,)
+    #   - (int[],)
+    #   - (int, int,)
+    #   - (int, int)
+    #   - (int, int, int,)
+    #   - (int, int, int,)
```

Comment:
Are these duplicates, or maybe missed removing the comma?

Reply:
Updated in 1b16442c